tst/EnergyPlus/unit/Furnaces.unit.cc

```diff
@@ -233,11 +237,102 @@ TEST_F(EnergyPlusFixture, SetVSHPAirFlowTest_VSFurnaceFlowTest)
+    WaterToAirHeatPumpSimple::SimpleWatertoAirHP(1).BranchNum = 1;
+    WaterToAirHeatPumpSimple::SimpleWatertoAirHP(1).CompNum = 1;
+
+    WaterToAirHeatPumpSimple::SimpleWatertoAirHP(2).AirInletNodeNum = 3;
+    WaterToAirHeatPumpSimple::SimpleWatertoAirHP(2).AirOutletNodeNum = 2;
+    WaterToAirHeatPumpSimple::SimpleWatertoAirHP(2).WaterInletNodeNum = 7;
+    WaterToAirHeatPumpSimple::SimpleWatertoAirHP(2).WaterOutletNodeNum = 8;
+    WaterToAirHeatPumpSimple::SimpleWatertoAirHP(2).LoopNum = 2;
+    WaterToAirHeatPumpSimple::SimpleWatertoAirHP(2).LoopSide = 1;
+    WaterToAirHeatPumpSimple::SimpleWatertoAirHP(2).BranchNum = 1;
+    WaterToAirHeatPumpSimple::SimpleWatertoAirHP(2).CompNum = 1;
+
+    // set up plant loop
+    DataPlant::TotNumLoops = 2;
+    DataPlant::PlantLoop.allocate(DataPlant::TotNumLoops);
+
+    for ( int loopindex = 1; loopindex <= DataPlant::TotNumLoops; ++loopindex ) {
+        auto &loop(DataPlant::PlantLoop(loopindex));
+        loop.LoopSide.allocate(2);
+        auto &loopside(DataPlant::PlantLoop(loopindex).LoopSide(1));
+        loopside.TotalBranches = 1;
+        loopside.Branch.allocate(1);
+        auto &loopsidebranch(DataPlant::PlantLoop(loopindex).LoopSide(1).Branch(1));
+        loopsidebranch.TotalComponents = 2;
+        loopsidebranch.Comp.allocate(2);
+    }
+    DataPlant::PlantLoop(1).Name = "Hot Water Loop";
+    DataPlant::PlantLoop(1).FluidName = "WATER";
+    DataPlant::PlantLoop(1).FluidIndex = 1;
+
+    DataPlant::PlantLoop(2).Name = "Chilled Water Loop";
+    DataPlant::PlantLoop(2).FluidName = "WATER";
+    DataPlant::PlantLoop(2).FluidIndex = 1;
+
+    DataPlant::PlantLoop(1).LoopSide(1).Branch(1).Comp(1).TypeOf_Num = DataPlant::TypeOf_CoilWAHPCoolingEquationFit;
+    DataPlant::PlantLoop(1).LoopSide(1).Branch(1).Comp(1).Name = "WATERCOOLINGCOIL";
+    DataPlant::PlantLoop(2).LoopSide(1).Branch(1).Comp(1).TypeOf_Num = DataPlant::TypeOf_CoilWAHPHeatingEquationFit;
+    DataPlant::PlantLoop(2).LoopSide(1).Branch(1).Comp(1).Name = "WATERHEATINGCOIL";
+
+    DataHeatBalance::HeatReclaimSimple_WAHPCoil.allocate(2);
+
+    WaterToAirHeatPumpSimple::GetCoilsInputFlag = false; // turn off water source coil GetInput
+    CalcNewZoneHeatCoolFlowRates(
+        FurnaceNum, firstHVACIteration, compOp, zoneLoad, moistureLoad, heatCoilLoad, reheatCoilLoad, onOffAirFlowRatio, hXUnitOn);
+    EXPECT_EQ(Furnace(1).MdotFurnace, 0.2); // flow rate is at idle speed flow rate
+    EXPECT_EQ(DataLoopNode::Node(1).MassFlowRate, 0.2); // furnace inlet node mass flow rate is at idle speed flow rate
+    EXPECT_EQ(Furnace(1).CoolPartLoadRatio, 0.0);
+    EXPECT_EQ(Furnace(1).HeatPartLoadRatio, 0.0);
+
+    Furnace(1).HeatPartLoadRatio = 1.0;
+    Furnaces::HeatingLoad = true;
+    CalcNewZoneHeatCoolFlowRates(
+        FurnaceNum, firstHVACIteration, compOp, zoneLoad, moistureLoad, heatCoilLoad, reheatCoilLoad, onOffAirFlowRatio, hXUnitOn);
+    EXPECT_EQ(Furnace(1).HeatPartLoadRatio, 1.0);
```

Comment:
This is as far as I got on the unit test. It runs and does test some code but there is more to do to set up coil capacity, etc., to get meaningful results. I'm just out of time.